src/poetry/mixology/partial_solution.py

```diff
@@ -213,6 +214,18 @@ def relation(self, term: Term) -> str:
         if positive is not None:
             return positive.relation(term)
 
+        positive = self._positive.get(term.dependency.name)
+        if positive is not None:
+            # TODO: replace deepcopy with dep.with_features() when
+            # https://github.com/python-poetry/poetry-core/pull/392 is available
+            dep = deepcopy(positive.dependency)
+            dep._features = frozenset(term.dependency.features)
+            # TODO: remove https://github.com/python-poetry/poetry-core/pull/375
+            dep._extras = frozenset(term.dependency.features)
+            positive_with_features = Term(dep, is_positive=True)
+            self._positive[term.dependency.complete_name] = positive_with_features
```

Comment:
this doesn't look like a method that I would expect to be making updates to `self`: is there some other moment at which this should have happened?

Reply:
Maybe, you're right. I am not completely satisfied with the solution myself. Will take another look at it.

Reply:
Moved it to `_register`. Looks better, doesn't it?

Reply:
Yes, this looks a much better place for it - thanks!

suggestion: we only need to look for `old_positive_without_features` if the assignment dependency _has_ features: else - which would usually be the case - we're just looking up the same thing twice.  So
```patch
         """
         name = assignment.dependency.complete_name
         old_positive = self._positive.get(name)
-        if old_positive is None:
+        if old_positive is None and assignment.dependency.features:
             old_positive_without_features = self._positive.get(
                 assignment.dependency.name
             )
```
?